.github/workflows/ci.yml

```diff
@@ -0,0 +1,30 @@
+name: "Continuous Integration"
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    types: [opened, synchronize, reopened]
+jobs:
+  continuous-integration:
+    name: continuous-integration
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    strategy:
+      matrix:
+        mocha-version: [3, 4, 5, 6, 7, 8, 9, 10]
+        node-version: [14, 16, 18, 19]
```

Comment:
@felipeplets do we need such extensive matrix, it'll take a lot of time, perhaps we can limit mocha to last 2 version and node to LTS and next?

Reply:
@piotrwitek have you tried running it? All of them will run in parallel and in my tests they run pretty fast.
Also, it is [free and we can run 20 in parallel](https://docs.github.com/en/actions/learn-github-actions/usage-limits-billing-and-administration). 

All we need is that you to approve the action below (you don't need to approve or merge this PR), it will run and we can see how long it takes and any limitations or issues.

This is definitely a conservative option, but the package.json currently says to support this all, so probably the best is not to remove the tests, but in the next major version remove the support of older versions. Because if the library supports the best option is to test it.

Reply:
Thanks for the clarification, I like how you evaluated benefits and drawbacks, it makes sense. Your advice is really helpful.